app/javascript/components/upload/ExpandableFileForm.jsx

```diff
@@ -205,17 +219,43 @@ function DeleteButton({ file, deleteFile, setShowConfirmDeleteModal }) {
  * @param {array} allFiles
  * @returns {boolean} Whether to show upload control buttons or not
  */
-function getIsUploadEnabled(isAnnDataExperience, allFiles, file) {
+function getIsUploadEnabled(isAnnDataExperience, file) {
   const isClustering = file.file_type === 'Cluster'
   const isExpressionMatrix = file.file_type === 'Expression Matrix'
-
   return !((isClustering || isExpressionMatrix) && isAnnDataExperience)
 }
 
+/**
+ * Determine whether to show the save/delete buttons.
+ *
+ * @param {boolean} isAnnDataExperience
+ * @param {array} allFiles
+ * @returns {boolean} Whether to show upload control buttons or not
+ */
+function getIsSaveEnabled(isAnnDataExperience, allFiles, file) {
+  // if it's not AnnDataExperience don't gate the save/delete buttons
+  if (!isAnnDataExperience) {return true}
+
+  // allow save/delete for updates after an AnnData file has been uploaded
+  let alreadyUploaded = false
+
+  alreadyUploaded = allFiles.some(fileObj => fileObj?.ann_data_file_info?.data_fragments.length > 0)
```

Comment:
Hmm that's a good thought. I'm not totally sure how this piece would work within the environment of multiple AnnData files. Using the logic as is with `every` and multiple AnnDatas would mean you couldn't save on any AnnData file if there is another AnnData file not yet uploaded but is being filled in. Whereas exact logic with `some` would allow saving on AnnDatas you are currently uploading as long as one is up. So both not quite right, with `some` being too permissive and `every` being too restrictive.

So I think instead of looking at `allFiles` when we go multi-AnnData we'd have to add some logic for each individual AnnData. 

So I think you are correct that `some` won't play right if keeping this exact logic when allowing multiple AnnData but I don't think `every` would work exactly right either and so either option will require some re-working to support multi.